CRM/Case/Form/Activity.php

```diff
@@ -203,6 +203,10 @@ public function setDefaultValues() {
       }
       $this->assign('targetContactValues', empty($targetContactValues) ? FALSE : $targetContactValues);
 
+      if ($this->_activityId) {
+        $this->_encounterMedium = CRM_Core_DAO::getFieldValue('CRM_Activity_DAO_Activity', $this->_activityId, 'medium_id');
```

Comment:
This is an undeclared property of the class so will cause warnings in php8. Why are we even setting it?

Reply:
hmm yep - previously the variable was being set from `buildQuickForm()`  and only used once in `setDefaultValues()`, so I moved it to `setDefaultValues`.

I updated my PR to simplify the code further. All these shenanigans around `encounterMediums` were odd, there is nothing special about this field.

Reply:
> shenanigans

Good word for it!